backend/services/applicationService.py

```diff
@@ -48,10 +48,12 @@ def updateApplication(userid, form):
     try:
         application = getApplicationByUserId(userid)
         for field in form.keys():
-            if(form[field]):
+            if(field in application.userEditableFields()):
                 setattr(application, field, form[field])
         db.session.add(application)
         db.session.commit()
+        if ("partnerUsername" in form.keys()):
```

Comment:
if(form["partnerUsername"]) is more correct. As it should not try to find a username if it's an empty string